matflow/param_classes/utils.py

```diff
@@ -39,3 +40,51 @@ def masked_array_from_list(
     if is_2D:
         out = out.reshape(n_rows, -1, order="C")
     return out
+
+
+def read_numeric_csv_file(
+    path: str,
+    number: int | None = None,
+    start_index: int = 0,
+    delimiter: str = " ",
+    columns: list[int] | None = None,
+    exc_msg: str | None = None,
+) -> NDArray:
+    """
+    Load data from a text file.
+
+    Parameters
+    ----------
+    path
+        Path to the file to load from.
+    number
+        Number of lines to read from the file.
+    start_index
+        The line number of the file that the seeds start at.
+        Allows skipping headers.
+    delimiter
+        The delimiter separating values in the file.
+        Defaults to space, but commas and tabs are also sensible
+        (and correspond to CSV and TSV files respectively).
+    columns
+        The columns in the file to read from.
+        Defaults to reading every column.
+    exc_msg:
+        ValueError message to return.
+    """
+    data: list[list[float]] = []
+    with Path(path).open("rt") as fh:
+        for idx, line in enumerate(fh):
+            line = line.strip()
+            if not line or idx < start_index:
+                continue
+            elif len(data) < number if number is not None else True:
+                values = line.split(delimiter)
+                if idx == 0:
```

Comment:
Just a small correction here to account for the case where we are not starting at the top of the file:
```suggestion
                if idx == start_idx:
```

Reply:
Ah good point!